// @hieuvu21/jobber-shared/build/src/logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.winstonLogger = void 0;
const tslib_1 = require("tslib");
const winston = tslib_1.__importStar(require("winston"));
const winston_elasticsearch_1 = require("winston-elasticsearch");
const esTransformer = (logData) => {
    return (0, winston_elasticsearch_1.ElasticsearchTransformer)(logData);
};
const winstonLogger = (elasticsearchNode, name, level) => {
    const consoleTransport = new winston.transports.Console({
        level,
        handleExceptions: true,
        format: winston.format.combine(winston.format.colorize(), winston.format.simple()),
    });
    const esTransport = new winston_elasticsearch_1.ElasticsearchTransport({
        level,
        transformer: esTransformer,
        clientOpts: {
            node: elasticsearchNode,
            maxRetries: 2,
            requestTimeout: 10000,
            sniffOnStart: false,
        },
    });
    const logger = winston.createLogger({
        exitOnError: false,
        defaultMeta: {
            service: name,
        },
        transports: [consoleTransport, esTransport],
    });
    return logger;
};
exports.winstonLogger = winstonLogger;
