// @hieuvu21/jobber-shared/build/error-handler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerError = exports.FileTooLargeError = exports.NotAuthorizeError = exports.NotFoundError = exports.BadRequestError = exports.customError = void 0;
const http_status_1 = require("http-status");
class customError extends Error {
    constructor(message, comingFrom) {
        super(message);
        this.comingFrom = comingFrom;
    }
    serializeError() {
        return {
            message: this.message,
            statusCode: this.statusCode,
            status: this.status,
            comingFrom: this.comingFrom,
        };
    }
}
exports.customError = customError;
class BadRequestError extends customError {
    constructor(message, comingFrom) {
        super(message, comingFrom);
        this.statusCode = http_status_1.status.BAD_REQUEST;
        this.status = 'error';
    }
}
exports.BadRequestError = BadRequestError;
class NotFoundError extends customError {
    constructor(message, comingFrom) {
        super(message, comingFrom);
        this.statusCode = http_status_1.status.NOT_FOUND;
        this.status = 'error';
    }
}
exports.NotFoundError = NotFoundError;
class NotAuthorizeError extends customError {
    constructor(message, comingFrom) {
        super(message, comingFrom);
        this.statusCode = http_status_1.status.UNAUTHORIZED;
        this.status = 'error';
    }
}
exports.NotAuthorizeError = NotAuthorizeError;
class FileTooLargeError extends customError {
    constructor(message, comingFrom) {
        super(message, comingFrom);
        this.statusCode = 413;
        this.status = 'error';
    }
}
exports.FileTooLargeError = FileTooLargeError;
class ServerError extends customError {
    constructor(message, comingFrom) {
        super(message, comingFrom);
        this.statusCode = http_status_1.status.SERVICE_UNAVAILABLE;
        this.status = 'error';
    }
}
exports.ServerError = ServerError;
