// @hieuvu21/jobber-shared/build/esm/error-handler.js
import { status } from 'http-status';
export class customError extends Error {
  constructor(message, comingFrom) {
    super(message);
    this.statusCode = void 0;
    this.status = void 0;
    this.comingFrom = void 0;
    this.comingFrom = comingFrom;
  }
  serializeError() {
    return {
      message: this.message,
      statusCode: this.statusCode,
      status: this.status,
      comingFrom: this.comingFrom
    };
  }
}
export class BadRequestError extends customError {
  constructor(message, comingFrom) {
    super(message, comingFrom);
    this.statusCode = status.BAD_REQUEST;
    this.status = 'error';
  }
}
export class NotFoundError extends customError {
  constructor(message, comingFrom) {
    super(message, comingFrom);
    this.statusCode = status.NOT_FOUND;
    this.status = 'error';
  }
}
export class NotAuthorizeError extends customError {
  constructor(message, comingFrom) {
    super(message, comingFrom);
    this.statusCode = status.UNAUTHORIZED;
    this.status = 'error';
  }
}
export class FileTooLargeError extends customError {
  constructor(message, comingFrom) {
    super(message, comingFrom);
    this.statusCode = 413;
    this.status = 'error';
  }
}
export class ServerError extends customError {
  constructor(message, comingFrom) {
    super(message, comingFrom);
    this.statusCode = status.SERVICE_UNAVAILABLE;
    this.status = 'error';
  }
}
