// @hieuvu21/jobber-shared/build/cjs/logger.js
"use strict";

exports.__esModule = true;
exports.winstonLogger = void 0;
var _winston = _interopRequireDefault(require("winston"));
var _winstonElasticsearch = require("winston-elasticsearch");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
const esTransformer = logData => {
  return (0, _winstonElasticsearch.ElasticsearchTransformer)(logData);
};
const winstonLogger = (elasticsearchNode, name, level) => {
  const consoleTransport = new _winston.default.transports.Console({
    level,
    handleExceptions: true,
    format: _winston.default.format.combine(_winston.default.format.colorize(), _winston.default.format.simple())
  });
  const esTransport = new _winstonElasticsearch.ElasticsearchTransport({
    level,
    transformer: esTransformer,
    clientOpts: {
      node: elasticsearchNode,
      maxRetries: 2,
      requestTimeout: 10000,
      sniffOnStart: false
    }
  });
  const logger = _winston.default.createLogger({
    exitOnError: false,
    defaultMeta: {
      service: name
    },
    transports: [consoleTransport, esTransport]
  });
  return logger;
};
exports.winstonLogger = winstonLogger;
